src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS/ProjectSystem/VS/Xproj/MigrateXprojProjectFactory.cs

```diff
@@ -150,6 +150,13 @@ internal HResult BackupAndDeleteGlobalJson(string solutionDirectory, string back
                 try
                 {
                     _fileSystem.RemoveFile(globalJson);
+
+                    var remover = new GlobalJsonRemover(_serviceProvider);
+                    HResult hr = solution.AdviseSolutionEvents(remover, out uint cookie);
```

Comment:
Yes. `IVsProjectUpgradeViaFactory` is an IVs interface, the invariant is that all calls must be made on the UI thread.